pkg/client/leaderelection/leaderelection.go

```diff
@@ -103,8 +103,8 @@ type LeaderElectionConfig struct {
 	// Identity is a unique identifier of the leader elector.
 	Identity string
 
-	EndpointsClient coreclientset.EndpointsGetter
-	EventRecorder   record.EventRecorder
+	Client        clientset.Interface
```

Comment:
I think that all over the code we are using clientset.Interface and it's good to be consistent.

Also, since @timothysc is going to switch the lock to something different than endpoints (some third party object), we will need a wider interface anyway.